src/pages/index.js

```diff
@@ -91,53 +75,106 @@ function renderCard(data) {
+
+const profileUpdateSubmit = new PopupWithConfirmation("#profile-edit-modal");
+const newCardSubmit = new PopupWithConfirmation("#add-new-card");
+
+function handleDeleteClick(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.setSubmitText(true, "Deleting...");
+    api
+      .deleteCard(card.id)
+      .then(() => {
+        deleteCardPopup.close();
+        card.removeCard();
+      })
+      .catch((err) => console.log(err))
+      .finally(() => deleteCardPopup.setSubmitText(false));
+  });
+}
 
+//FORM POPUP
 const openImagePopup = new PopupWithForm(
   "#add-new-card",
   handleImageFormSubmit
 );
-const editProfile = new PopupWithForm(
-  "#profile-edit-modal",
-  handleFormSubmit
+const editProfile = new PopupWithForm("#profile-edit-modal", handleFormSubmit);
+const profilePicEdit = new PopupWithForm(
+  "#profile__image-edit",
+  profilePicEditSubmit
 );
+
 editProfile.setEventListeners();
+profilePicEdit.setEventListeners();
 openImagePopup.setEventListeners();
 
 const editFormValidator = new FormValidator(config, profileModalForm);
+const profilePicValidator = new FormValidator(config, profilePicModal);
 const cardFormValidator = new FormValidator(config, addNewCardModal);
 
 editFormValidator.enableValidation();
+profilePicValidator.enableValidation();
 cardFormValidator.enableValidation();
 
 //FORM SUBMIT
 function handleFormSubmit(data) {
-  console.log(data)
-  userInfo.setUserInfo(data)
-  editProfile.close();
+  api.profileUpdate(data).then(() => {
+    userInfo.setUserInfo(data);
+    profileUpdateSubmit.setSubmitText(true, "Saving...");
+    editProfile.close();
+  });
 }
 
 // //IMAGE SUBMIT
 function handleImageFormSubmit() {
   const name = newCardModalTitle.value;
   const link = newCardModalUrl.value;
-  generateCard({ name, link });
-  openImagePopup.close();
+  api.addCard(name, link).then((card) => {
+    generateCard(card);
+    newCardSubmit.setSubmitText(true, "Saving...");
+    openImagePopup.close();
+  });
+}
+
+function profilePicEditSubmit() {
+  const url = profilePicUrl.value;
+  api.updateProfileAvatar(url).then(() => {
+    profileImage.src = url;
+    profilePicEdit.close();
+  });
 }
 
+//IMAGE LIKE
+
 //PROFILE EDIT POPUP
 const userInfo = new UserInfo(".profile__title", ".profile__description");
 
-//EDIT PROFILE
+//EVENT LISTENERS
 profileEditButton.addEventListener("click", () => {
   profileModalName.value = profileName.textContent;
   profileModalDescription.value = profileDescription.textContent.trim();
   editProfile.open();
 });
 
-//EVENT LISTENERS
 addNewCardButton.addEventListener("click", () => {
   cardFormValidator.toggleButtonState();
   openImagePopup.open();
 });
 
+profilePicButton.addEventListener("click", () => {
+  profilePicEdit.open();
+});
+
+//API CALLS
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([data, cards]) => {
+    userInfo.setUserInfo({ description: data.about, title: data.name });
```

Comment:
* When I change the avatar and reload the page the old avatar appears with Cousteau